Base Test Coverage and Test Organization

marty-larocque opened the pull request:
This pull request does a number of things.

- Adds some tests for the base module, mostly for `assert_xydata()`, addressing issue #101 
- Renames tests that shared names with other tests, causing only one to be defined and run. This addresses issue #101 
- Refactors `assert_xydata()` for clarity
- Moves the test `test_correct_title()` to test_base_titles_captions.py, addressing issue #115 
- Moves all the tests for `assert_xydata()` to test_base_data.py, addressing issue #114

marty-larocque committed:
Adding test coverage, working with datetime.

marty-larocque committed:
Adding Tests

lwasser commented on matplotcheck/tests/test_base.py:
```diff
@@ -68,21 +72,62 @@ def test_assert_xydata_scatter(pt_scatter_plt, pd_df):
 
 
-def test_assert_xydata_scatter(pt_scatter_plt, pd_df):
+def test_assert_xydata_tolerance(pt_scatter_plt, pd_df):
```
hey @ryla5068 please add docstrings to your function tests!

marty-larocque committed:
Uploading for CodeCov report

marty-larocque committed:
Added tests for tolerance

marty-larocque committed:
Moved some tests around, added test_base_data file, added docstrings to tests.

marty-larocque committed:
Removed PDB from imports for commit

marty-larocque commented:
@lwasser Could you take a look at this?

marty-larocque committed:
Added Test

marty-larocque committed:
Fixed error

lwasser commented on matplotcheck/base.py:
```diff
@@ -675,50 +678,58 @@ def assert_xydata(
+            for i in range(100):
                 np.testing.assert_allclose(
-                    xy_data["x"], xy_expected[xcol], rtol=tolerence, err_msg=m
+                    xy_data.iloc[0, 0:i], xy_expected.iloc[0, 0:i], rtol=tolerence, err_msg=m
```
hey @ryla5068 can you explain why some of these changes were implemented? what is hard for me to see if why the change was made given many of the tests are new! i'm guessing it's because pandas requires iloc now to call cols and rows but here you are using sequential calls rather than the entire column

lwasser commented on matplotcheck/base.py:
```diff
@@ -675,50 +678,58 @@ def assert_xydata(
+            if xtime:
+                raise ValueError("tolerance must be 0 with datetime on x-axis")
+            """
+            for i in range(100):
```
where did the value of 100 come from for this for loop and why did you implement a loop here?

lwasser replied:
resolved with the comment below!

lwasser commented on matplotcheck/base.py:
```diff
@@ -675,50 +678,58 @@ def assert_xydata(
                 )
                 np.testing.assert_allclose(
-                    xy_data["y"], xy_expected[ycol], rtol=tolerence, err_msg=m
+                    xy_data.iloc[1, 0:i], xy_expected.iloc[1, 0:i], rtol=tolerence, err_msg=m
```
similar question here. i'm not sure why there is a loop and what the value of 100 represents.

lwasser commented on matplotcheck/base.py:
```diff
@@ -675,50 +678,58 @@ def assert_xydata(
             Nothing if no legends overlap, otherwise throws ``AssertionError`` with message `m`
         """
+        if xy_expected is None:
+            return
```
Can you explain using return vs pass here?

marty-larocque replied:
The expected behavior is that the assertion passes if `xy_expected` is `None`.

In a place like this, `pass` is effectively an empty line. If it remains unchanged, and `xy_expected` is `None`, then the function continues to try to evaluate `xy_expected` like it contains data, and eventually throws an error.

By replacing it with `return`, the assertion passes as expected.

marty-larocque committed:
Removed comment

marty-larocque commented:
@lwasser The for loop and .iloc[] were temporary things I'd put in for debugging. I'm pretty confused how they ended up in the most recent commit, as I deleted them a couple commits ago, but they should be gone now.

lwasser commented on matplotcheck/base.py:
```diff
@@ -15,6 +15,7 @@
 from scipy import stats
 import pandas as pd
 import geopandas as gpd
+import datetime
```
interesting. it datetime called explicitly in this module @ryla5068 ? i know when it inherits it will be required but that is in timeseries. i'm just curious how it was working previously!

marty-larocque replied:
This was left in from when I was trying to convert times in the base module. I just pushed a commit removing it.

lwasser replied:
great!! thanks @ryla5068 !!!

marty-larocque committed:
Removed datetime module

lwasser commented on matplotcheck/base.py:
```diff
@@ -675,51 +677,50 @@ def assert_xydata(
+        # If xy_expected is a GeoDataFrame, then we make is a normal DataFrame
+        # with the coordinates of the geometry in that GeoDataFrame as the x and
+        # y data
+        if isinstance(xy_expected, gpd.geodataframe.GeoDataFrame) and not xcol:
```
This is an excellent improvement. (isinstance vs ==)

lwasser commented on matplotcheck/tests/conftest.py:
```diff
@@ -25,6 +26,27 @@ def pd_df_timeseries():
+    return gdf
+
+
+@pytest.fixture
```
@ryla5068 will you please add docstrings to all fixtures so we know what they are for. it can be a short docstring (single line)

lwasser commented on matplotcheck/tests/test_base.py:
```diff
@@ -51,41 +51,9 @@ def test_options(pt_line_plt):
-def test_assert_xydata_xlabel(pt_bar_plt, pd_df):
-    pd_df["A"] = pd_df["A"].apply(str)
-    pt_bar_plt.assert_xlabel_ydata(pd_df, xcol="A", ycol="B")
+"""
```
ahhhh this test should go in a spatial tests file. @ryla5068 why don't you create the file and add this even if it's commented out ... this way it's not left in base.

lwasser replied:
also @ryla5068 please see above

https://github.com/earthlab/matplotcheck/pull/112/files#diff-8b07e1c1007f53de8610db3fec8039f7R44 
there is a question on line 44 that i didn't notice. can you kindly look into that? if it's not your question, then i think we can just remove it. we did decide to write individual tests which will be easier to manage and fix over time.

lwasser reviewed (changes requested):
ok @ryla5068 over all you've made some great improvements to the code.

please see my comments. also

1. anytime we use pytest.raises() please add `match=""`. essentially we want to ensure that the error raised also has the message that we expect. you can find examples of this within the code as well!

example that is missing match= :  
https://github.com/earthlab/matplotcheck/pull/112/files#diff-8b07e1c1007f53de8610db3fec8039f7R23

- [ ] you can also move the timeseries test that is in test_base_data.py to a new file test_ts_data.py even if it's commented out. let's keep time series tests together [(lines 58-64)](https://github.com/earthlab/matplotcheck/pull/112/files#diff-79817b1c10a954d0ba549e86b352fc49R58-R64)

2. please add docstrings to fixtures. they can be very short like single line

the last thing that i'm just noting. Black is running but not actually checking files properly. i'm not sure why that is happening but might create an issue!! i see many formatting issues in these files. but that is something that is wrong with our dev envt!!

lwasser commented:
ok @ryla5068 this all looks great. i ddi some testing of black locally. for some reason it's not formatting the docstrings using the 79 character line length. i'm not sure why but let's just ensure the code is 79 characters is less long including the docstrings. i'll create an issue surrounding this but you can adjust the code manually for this pr. once you are done with the fixes, please ping me here again and i'll give this another review (and will likely merge as my comments are very small!!)

lwasser commented:
a few other question

- [ ] https://codecov.io/gh/earthlab/matplotcheck/pull/112/diff?src=pr&el=tree#D3-683 i am wondering if the test for this doesn't have match= to rest that the error message is raised?

marty-larocque committed:
Some final touches

marty-larocque committed:
Merge branch 'master' into base_test_coverage

marty-larocque committed:
Fixed arror causing documentation build error

marty-larocque committed:
Moved timeseries test to test_ts_data.py

marty-larocque commented:
@lwasser I think I've fixed everything here. 
- For the commented-out tests:
     - The geodata test was not a working or substantial test, so I deleted it.
     - I moved the timeseries test to `test_ts_data.py`, but left it commented out.
- I couldn't follow the link you gave to the `pytest.raises()` call without a `match=`, but I added a `match=` to a couple tests in `test_base.py`. I believe every test in that file and in`test_base_data.py` has a `match=` now.
- I added docstrings to the fixtures
- I changed all the docstrings in `base.py` to have the correct length lines
- Regarding the codecov link you sent me (`base.py:683`), that line just does not have a test right now. That should be addressed in a future pr.

lwasser commented on matplotcheck/tests/test_base_data.py:
```diff
@@ -0,0 +1,87 @@
+    plt.close()
+
+
+'''
```
@ryla5068 super minor. will you please move this code OUT of the test_base_data.py file and create a new test_base_timeseries_data.py file? it's ok if it remains commented out id just like it in a new file.

lwasser commented on matplotcheck/tests/test_ts_data.py:
```diff
@@ -0,0 +1,9 @@
+import pytest
+
+'''
```
@ryla5068 ok you moved this. i'm going to delete it from the base module now.

lwasser reviewed (approved):
i'm going to make a few changes and then merge this @ryla5068 . the time series code just never was deleted from base but it was moved to the new test_ts_data.py file! .

Commit:
removing duplicate time series code

lwasser commented:
excellent work on this PR @ryla5068 !!! i'm going to let CI run and will then merge this.
